### DISCUSSION TITLE
How to retrieve the current data collection displayed from a filtered DetailList control

### DISCUSSION THREAD

**Discussion by AstroNorman** [microsoft/powercat-creator-kit#76] [Q&A] [answered]
<img width="468" alt="detaillist" src="https://user-images.githubusercontent.com/106515086/180912659-4b5dea16-49a5-4c96-b4bc-7b96a080296a.PNG">

I have a DetailList control and I have added the ability to sort and search for specific item/items in the table.

Right now, I plan on exporting the data displayed in the table to an Excel file using power automate.

In order to do so, I will need to be able to retrieved the current sorted/filtered data shown in the DetailList control. 

How do I to retrieved the current sorted / filtered data collections from the DetailList control?

**Comment by Ramakrishnan24689** [comment, marked as answer]
Hi @AstroNorman,

As the dataset(Collection) is filtered outside of Details list, one way to pass the collection which is filtered/Sorted to a PowerAutomate for generating an excel etc. can be to re-use the same PowerFx.

For example, if below code is used to modify collection which is passed as an **Items** property of detail list. Then same code can be re-used and passed as input parameter for the PowerAutomate to generate excel or perform any other task.
```
SortByColumns(
        Search(
            colData,
            CK_SearchBox.SearchText,
            "ColumnName"
        )
    ,
    ctxSortCol,
    If(
        ctxSortAsc,
        SortOrder.Ascending,
        SortOrder.Descending
    )
)
```

Same can be re-used like

```
'Export2Excel'.Run(JSON(SortByColumns(
        Search(
            colData,
            CK_SearchBox.SearchText,
            "ColumnName"
        )
    ,
    ctxSortCol,
    If(
        ctxSortAsc,
        SortOrder.Ascending,
        SortOrder.Descending
    )
))),JSONFormat.IncludeBinaryData))
```
Hope this helps.
